src/router.js

```diff
@@ -10,6 +10,25 @@ const configUrl = process.env.CONFIG_BACKEND_URL || 'https://api.rudderstack.com
 const jsSdkCdnUrl =
   process.env.JS_SDK_CDN || 'https://cdn.rudderlabs.com/v1.1/rudder-analytics.min.js';
 
+const ensureHttpsPrefix = (url) => {
+  // Check if the URL starts with http:// or https://
+  if (!/^https?:\/\//i.test(url)) {
+    return `https://${url}`;
+  }
+  return url;
+};
+
+const isValidDataPlaneURL = (dataPlaneUrl) => {
+  const newDataPlaneUrl = ensureHttpsPrefix(dataPlaneUrl);
+  try {
+    new URL(newDataPlaneUrl); // This will throw if the URL is invalid
+    return true;
+  } catch {
+    return false;
+  }
+};
+const isValidWriteKey = (writeKey) => /^[A-Za-z0-9_]{5,}$/.test(writeKey);
```

Comment:
I didnt get this regex - what is it testing?

Reply:
it is testing that any alpha numeric with minimum 5 chars

Reply:
ok